tcmur_device.c

```diff
@@ -74,8 +74,10 @@ int __tcmu_reopen_dev(struct tcmu_device *dev, bool in_lock_thread, int retries)
 		rdev->lock_state = TCMUR_DEV_LOCK_UNLOCKED;
 	pthread_mutex_unlock(&rdev->state_lock);
 
-	tcmu_dev_dbg(dev, "Closing device.\n");
-	rhandler->close(dev);
+	if (rdev->flags & TCMUR_DEV_FLAG_IS_OPEN) {
```

Comment:
Is it safe to test the flags while not holding the lock?

Reply:
It is because the places it can change is a couple lines below the line above and then in tcmur_stop_device which can only run when everything is shutdown.

I will change it just to be consistent because it does look odd.